### ISSUE TITLE
Unsound example and missing safety docs

### ISSUE THREAD

**Issue by niluxv** [bitvecto-rs/bitvec#147]
The doctest for `BitVec::as_mut_bitptr` is unsound as is reported (i think) correctly by `miri`.
The example is the following:
```rust
use bitvec::prelude::*;

let mut bv = BitVec::<Msb0, u8>::with_capacity(4);
let bp = bv.as_mut_bitptr();
unsafe {
  for i in 0 .. 4 {
    bp.add(i).write(true);
  }
  bv.set_len(4);
}
assert_eq!(bv, bits![1; 4]);
```
This creates an unititialised `BitVec` `bv`. Then a mutable pointer to the allocation is derived and `bv` is initialised bit-by-bit. But setting a bit results in a `fetch_or` or `fetch_and` call on `Cell<u8>` but this is UB for an uninitialised byte.

The safety documentation for `BitPtr` also doesn't mention the fact that the pointed-to memory needs to be initialised for writes to be sound.

**Comment by myrrlyn** [comment]
... This is what I get for copying the stdlib doc text and tests without thinking about it too much.

This doctest doesn't exist in the v1 series, and I'll be sure to keep it gone. Thank you for bringing it to my attention. I don't run Miri on doctests (last I checked, I think it was unable, and I've never actually copied doctests into unit tests to run them through it), so that's on me.

I've updated the documentation on `BitVec`, `BitPtr`, and `BitAccess` accordingly.
